src/main/java/mappers/CsvValidator.java

```diff
@@ -74,70 +74,82 @@ public CsvValidator(String nuxeoUrl, String nuxeoUser, String nuxeoPassword, Str
     }
 
     public HashMap<String, ArrayList<String>> validate(String path, int limit) throws IOException{
-
         String header[] = csvReader.readNext();
         String fileTypes[] = {"AUDIO", "VIDEO", "IMG"};
         String nextLine[];
         Map<String, Integer> files_read = new HashMap<>();
         int lineNumber = 0;
-        String headerTemp;
+        String columnHeader;
+
+        Map<String, String[]> rowsProcessed = new TreeMap<String, String[]>();
 
-        while((nextLine = csvReader.readNext()) != null){
+        while((nextLine = csvReader.readNext()) != null) {
             files_read.clear();
-            int wordCount=0;
+            int columnCount=0;
             lineNumber++;
-            for (String word: nextLine) {
 
-                headerTemp= header[wordCount];
+            // Add to rows processed if not a duplicate
+            if (!rowsProcessed.containsKey(nextLine[0])) {
+                rowsProcessed.put(nextLine[0], nextLine);
+            }
+            // This is a duplicate within the CSV file - mark as such
+            else {
+                addToInvalid("Duplicates", "Cannot upload duplicate words in CSV: line " + (lineNumber + 1) +", " + nextLine[0]);
+            }
+
+            // Review column values for each row
+            for (String column : nextLine) {
+
+                columnHeader = header[columnCount];
 
                 for (String type: fileTypes) {
-                    if(header[wordCount].startsWith(type))
-                        headerTemp = header[wordCount].substring(type.length());
+                    if(header[columnCount].startsWith(type))
+                        columnHeader = header[columnCount].substring(type.length());
                 }
 
-// Disable if duplicate words need to be added as well as WordMapper line 123 and below
-                if(headerTemp.equals("WORD"))
-                    checkWordDuplicate(word, lineNumber);
+                // Disable if duplicate words need to be added as well as WordMapper line 123 and below
+                // This checks for duplicates against the remote DB, not within the CSV
+//                if(columnHeader.equals("WORD"))
+//                    checkWordDuplicate(column, lineNumber);
```

Comment:
This should probably be uncommented/enabled by default.
```suggestion
                if(columnHeader.equals("WORD"))
                    checkWordDuplicate(column, lineNumber);
```